qiskit_nature/second_q/operators/vibrational_op.py

```diff
@@ -265,11 +272,61 @@ def _validate_keys(self, keys: Collection[str]) -> None:
 
     @classmethod
     def _validate_polynomial_tensor_key(cls, keys: Collection[str]) -> None:
-        ...
+        allowed_chars = {"+", "-", "_"}
+
+        for key in keys:
+            if set(key) != allowed_chars:
+                raise QiskitNatureError(
+                    f"The key {key} is invalid. PolynomialTensor keys must be multiples of the "
+                    "`_+-` character sequence, for them to be expandable into a VibrationalOp.\n"
+                    "But this key contains other characters."
+                )
+            if len(key) % 3:
+                raise QiskitNatureError(
+                    f"The key {key} is invalid. PolynomialTensor keys must be multiples of the "
+                    "`_+-` character sequence, for them to be expandable into a VibrationalOp.\n"
+                    "But the length of this key is not a multiple of 3."
+                )
+            if key != "_+-" * (len(key) // 3):
+                raise QiskitNatureError(
+                    f"The key {key} is invalid. PolynomialTensor keys must be multiples of the "
+                    "`_+-` character sequence, for them to be expandable into a VibrationalOp."
+                )
 
     @classmethod
     def from_polynomial_tensor(cls, tensor: PolynomialTensor) -> VibrationalOp:
-        ...
+        cls._validate_polynomial_tensor_key(tensor.keys())
+
+        data: dict[str, _TCoeff] = {}
+
+        def _reshape_index(index):
+            new_index = []
+            for idx in range(0, len(index), 3):
+                new_index.extend([index[idx], index[idx + 1], index[idx], index[idx + 2]])
+            return new_index
+
+        for key in tensor:
+            if key == "":
+                # TODO: deal with complexity
```

Comment:
This is for the future. The same todo exists in the other operator classes. This goes back to that discussion we once had around how we could deal with stack-wide `complex` vs `float` dtypes and if/where this would really be necessary

I guess I should open a separate issue to track this. I will look into summarizing our previous discussions around this tomorrow, too :+1:

Reply:
Ah yes, that comment is phrased a bit poorly in that regard..